asm/instruction.go

```diff
@@ -107,7 +114,7 @@ func (ins Instruction) Marshal(w io.Writer, bo binary.ByteOrder) (uint64, error)
 	}
 
 	data := make([]byte, InstructionSize)
-	data[0] = byte(ins.OpCode)
+	data[0] = byte(ins.OpCode & 0x00FF)
```

Comment:
Doesn't this miss setting ins.Offset if this is SDiv, SMod?

Nit: magic number, probably better as a constant?